File: monsoon3.0/src/main/java/wtf/monsoon/api/util/misc/StringUtil.java
package wtf.monsoon.api.util.misc;

import com.github.javafaker.Faker;
import org.luaj.vm2.ast.Str;

import java.util.Random;

/**
 * @author Surge
 * @since 30/07/2022
 */
public class StringUtil {

    public static String formatEnum(Enum<?> enumIn) {
        String text = enumIn.toString();
        StringBuilder formatted = new StringBuilder();

        try {
            if(enumIn.getClass().getMethod("toString").getDeclaringClass().toString().contains("monsoon")) {
                return text;
            }
        } catch (Exception exception) {
            exception.printStackTrace();
        }

        if(text.equalsIgnoreCase("ncp")) return "NCP";

        int index = 0;
        boolean isFirst = true, isNewWord = false;
        for (char c : text.toCharArray()) {
            if (c == '_') {
                isNewWord = true;
                continue;
            }

            if (isFirst) {
                if(String.valueOf(c).toLowerCase().equals(String.valueOf(c))) formatted.append(c);
                else formatted.append(String.valueOf(c).toUpperCase());
                isFirst = false;
            } else if(isNewWord) {
                if(String.valueOf(c).toLowerCase().equals(String.valueOf(c))) formatted.append(" ").append(c);
                else formatted.append(String.valueOf(c).toUpperCase());
                isNewWord = false;
            } else {
                formatted.append(String.valueOf(c).toLowerCase());
            }

            index++;
        }

        return formatted.toString();
    }

    public static String getRandomString(int length) {
        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
        StringBuilder salt = new StringBuilder();
        Random rnd = new Random();
        while (salt.length() < length) {
            int index = (int) (rnd.nextFloat() * characters.length());
            salt.append(characters.charAt(index));
        }
        return salt.toString();
    }

    public static String getValidUsername() {
        Faker faker = new Faker();
        String username = faker.superhero().prefix() + faker.name().firstName() + faker.address().buildingNumber();
        System.out.println(username);
        return username;
    }

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/util/misc/StringUtil.kt
package wtf.monsoon.client.util.misc

import com.github.javafaker.Faker
import java.util.*

/**
 * @author Surge
 * @since 30/07/2022
 */
object StringUtil {
    fun formatEnum(enumIn: Enum<*>): String {
        val text = enumIn.toString()
        val formatted = StringBuilder()
        try {
            if (enumIn.javaClass.getMethod("toString").declaringClass.toString().contains("monsoon")) {
                return text
            }
        } catch (exception: Exception) {
            exception.printStackTrace()
        }
        if (text.equals("ncp", ignoreCase = true)) return "NCP"
        var index = 0
        var isFirst = true
        var isNewWord = false
        for (c in text.toCharArray()) {
            if (c == '_') {
                isNewWord = true
                continue
            }
            if (isFirst) {
                if (c.toString()
                        .lowercase(Locale.getDefault()) == c.toString()
                ) formatted.append(c) else formatted.append(
                    c.toString().uppercase(
                        Locale.getDefault()
                    )
                )
                isFirst = false
            } else if (isNewWord) {
                if (c.toString().lowercase(Locale.getDefault()) == c.toString()) formatted.append(" ")
                    .append(c) else formatted.append(
                    c.toString().uppercase(
                        Locale.getDefault()
                    )
                )
                isNewWord = false
            } else {
                formatted.append(c.toString().lowercase(Locale.getDefault()))
            }
            index++
        }
        return formatted.toString()
    }

    fun getRandomString(length: Int): String {
        val characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890"
        val salt = StringBuilder()
        val rnd = Random()
        while (salt.length < length) {
            val index = (rnd.nextFloat() * characters.length).toInt()
            salt.append(characters[index])
        }
        return salt.toString()
    }

    val validUsername: String
        get() {
            val faker = Faker()
            val username = faker.superhero().prefix() + faker.name().firstName() + faker.address().buildingNumber()
            println(username)
            return username
        }
}